### ISSUE TITLE
Plone 5.2.2 - Navigating within the folder contents does not update the URL of the actions in the toolbar

### ISSUE THREAD

**Issue by mcasagrande18** [plone/Products.CMFPlone#3191]
In Plone 5.2.2 when you navigation using the folder contents view through a hierarchy of documents then URL of the toolbar's view, edit, content and other action does not relate to the URL of the current folder of the folder contents view.

Is related to: https://github.com/plone/Products.CMFPlone/issues/2395 and https://github.com/plone/Products.CMFPlone/issues/2531

**Comment by fredvd** [comment]
I can confirm this. Noticed it in a project with a customised theme, my colleague @mauritsvanrees tested and sees the same in in coredev-plone5.2

**Comment by fredvd** [comment]
Did some more 'black box testing with creating a folder1/subfolder2/subsubfolder3/subsubsubfolder4 hierarchy. 

What happens is that only the FIRST navigation action in the folder_contents is pushed to the toolbar.  subsequent navigation actions in the folder contents by clicking on an item in the main table or the breadcrumbs is not. 

So when you enter folder_contents in subsubfolder3 and navigate to subfolder2 using the folder_contents breadcrumbs,  the view action in the toolbar is updates to /folder1/subfolder2, but after this it stays stuck on folder1/subfolder2

**Comment by fredvd** [comment]
Fix: https://github.com/plone/mockup/pull/1028

**Comment by fredvd** [comment]
Fixed in pending Plone 5.2.3 in mockup 3.2.3./plone.staticresources 1.3.0 .   Pull request for Plone 6 mockup: https://github.com/plone/mockup/pull/1030

**Comment by fredvd** [comment]
Unfortunately this doesn't seem fixed in the releases plone.staticresources 1.4.0  It's even worse now: the toolbar doesn't even get updated the first time.  I've tested this in a customer project on a testing environment where I pinned plone.staticresources to 1.4.0 which should contain the fix from mockup 3.2.3 .

Will investigate further, something is def. different with the javascript when being served from source files vs being served from bundles or metabundled for production. 

@mauritsvanrees  I cheered too early for a fix in 5.2.3 :-(

**Comment by fredvd** [comment]
I found the cause why the toolbar-updating completely broke in one of our customer projects.  We are using backend.xml there to switch the theme to barceloneta on edit/control panel pages. 

About two years ago @thet updated the main rules.xml in barceloneta to copy #edit-bar from the content and not the content-children of edit bar.  But he forgot to update the backend.xml version::
https://github.com/plone/plonetheme.barceloneta/commit/8beb307dd8968c0c58af2cf9727c7e08325cf729

both in plonetheme.barceloneta 2.2 and master when you add backend.xml, the div with #edit-bar disappears. 

When I tried to fix the navigation/toolbar sync issue  last week I worked with buildout-coredev 5.2 and included a dependency on `#edit-bar` to be sure to find the correct edit-zone to replace:

https://github.com/plone/mockup/blob/2a7c5707ecebbf601ed13d5b2ab12485870949a8/mockup/patterns/toolbar/pattern.js#L511-L522

At the moment Plone 5.2.3 is going to break toolbar/folder_contents in projects who are using the backend.xml system from plonetheme.barceloneta.  As what I saw in my project, the toolbar isn't updated at all because `#edit-bar` doesn't exist.

To avoid this, replacing `#edit-bar` with `body` in

https://github.com/plone/mockup/blob/2a7c5707ecebbf601ed13d5b2ab12485870949a8/mockup/patterns/toolbar/pattern.js#L520

Fixes the issue for backend.xml not copying `#edit-bar`.  Will need a new mockup/plone.staticresources release.  And we should fix backend.xml in plonetheme.barceloneta.
